### ISSUE TITLE
Moving Average problem 

### ISSUE THREAD

**Issue by MaliceDee** [316k/midi-ribbon-controller#1]
The main sketch won't compile and being fairly new to all this I'm having a hard time working out exactly why, when i try to compile the sketch I keep getting the following error message;
 `In file included from C:\Users\pixie\Documents\Arduino\libraries\midi-ribbon-controller-master\midi-ribbon-controller\midi-ribbon-controller.ino:3:0:

C:\Users\pixie\Documents\Arduino\libraries\midi-ribbon-controller-master/moving_average.h:49:13: error: macro "max" requires 2 arguments, but only 1 given

     int max() {

             ^

C:\Users\pixie\Documents\Arduino\libraries\midi-ribbon-controller-master/moving_average.h:62:13: error: macro "min" requires 2 arguments, but only 1 given

     int min() {

             ^`
I've looked through the moving_average.h file to try and see if I can work out why I'm getting this error but I haven't been able to crack this one, everything else here looks spot on and I reverse engineered it without the moving average functions and got it working but I'd like to understand how to fix this issue and implement a way to average the readings. 
Any help or advice will be greatly appreciated! Thank you in advance for your time 
Warm Regards,

**Comment by 316k** [comment]
Hi!

Thanks a lot for the interest in the projet, although I have to admit I haven't touched this in a while...

It seems like min and max are now keywords defined in the Arduino API : https://www.arduino.cc/reference/en/language/functions/math/max/, which wasn't the case a few years ago when I worked on this.

If this is the only problem, the solution should be pretty simple : rename the two functions to something like minValue() and maxValue(). I looked through the code to see what to refactor, but it seems they never are actually used...

I think simply deleting these two functions will work. Remove everything from line 49 to 71 in moving_average.h, I hope it will resolve your problem :)
